tests/commands/test_uninstall.py

```diff
@@ -0,0 +1,222 @@
+def test_remove_file_unlink_error(mock_unlink, tmp_path: pathlib.Path):
+    """Test successful remove_file if the file does not exist."""
+    test_file = tmp_path / "test_file"
+
+    test_file.write_text("test content")
+    mock_unlink.side_effect = Exception("unknown_exception")
+    assert not uninstall.remove_file(test_file)
+
+
+@mock.patch("pathlib.Path.exists")
+@mock.patch("pathlib.Path.unlink")
+def test_remove_services(mock_unlink, mock_exists, tmp_path: pathlib.Path, monkeypatch):
+    """Test remove_services invokes expected filesystem calls."""
+    # Let's create a systemd generated services directory
+    # so the system thinks there are services to delete there too.
+    generated_services_dir = tmp_path / "generator"
+    monkeypatch.setattr(
+        "quipucordsctl.commands.install.settings.SYSTEMD_GENERATED_SERVICES_DIR",
+        generated_services_dir,
+    )
+
+    # Make sure the system thinks all service files exist.
+    mock_exists.return_value = True
+
+    assert uninstall.remove_services()
+
+    assert mock_unlink.call_count == len(settings.TEMPLATE_SERVER_ENV_FILENAMES) + len(
+        settings.TEMPLATE_SYSTEMD_UNITS_FILENAMES
+    ) + len(settings.SYSTEMD_SERVICE_FILENAMES)
+
+
+def test_reload_daemon(mock_shell_utils):
+    """Test reload_daemon invokes expected shell commands."""
+    assert uninstall.reload_daemon()
+    mock_shell_utils.run_command.assert_has_calls(
+        (
+            mock.call(settings.SYSTEMCTL_USER_RESET_FAILED_CMD),
+            mock.call(settings.SYSTEMCTL_USER_DAEMON_RELOAD_CMD),
+        )
+    )
+
+
+@mock.patch("quipucordsctl.commands.uninstall.shutil.rmtree")
+def test_remove_data(mock_rmtree):
+    """Test remove data invokes the expected shell utilities commands."""
+    uninstall.remove_data()
+    expected_calls = []
+    for data_dir in settings.SERVER_DATA_SUBDIRS_EXCLUDING_DB.values():
+        expected_calls += [mock.call(data_dir, ignore_errors=True)]
+    mock_rmtree.assert_has_calls(expected_calls)
+
+
+def test_remove_secrets():
+    """Test removes secrets invokes the expected Podman utilities commands."""
+    with mock.patch.object(uninstall, "podman_utils") as mock_podman_utils:
+        mock_podman_utils.delete_secret.return_value = True
+
+        assert uninstall.remove_secrets()
+
+        for key in settings.QUIPUCORDS_SECRET_KEYS:
+            mock_podman_utils.delete_secret.assert_any_call(key)
+
+
+def test_remove_secrets_failure():
+    """Test function fails if one secret could not be removed."""
+    with mock.patch.object(uninstall, "podman_utils") as mock_podman_utils:
+        mock_podman_utils.delete_secret.return_value = False
+
+        assert not uninstall.remove_secrets()
+
+        key = list(settings.QUIPUCORDS_SECRET_KEYS)[0]
+        mock_podman_utils.delete_secret.assert_any_call(key)
+
+
+def test_uninstall_run(capsys):
+    """Test the command invokes the expected subcommands."""
+    mock_args = mock.Mock()
```

Comment:
If you accept my earlier recommended change that adds `if not args.quiet:` then using `mock.Mock()` as the `args` might not behave as you expect. In my own PR, I've been replacing Mocks like this with a _real_ `Namespace` object which is the correct type that argparse would return and because it doesn't return more funky child `Mocks` when queried for attributes that should otherwise be falsy.
```suggestion
    mock_args = argparse.Namespace()
```
An empty one like this works just fine most of the time. It's a lot cleaner than trying to fake child args and/or manipulate the hasattr/getattr helpers.

You may need to add an `import argparse` up top for this.